pykokkos/interface/type_inference.py

```diff
@@ -0,0 +1,234 @@
+    if isinstance(policy, int):
+        policy = RangePolicy(km.get_default_space(), 0, policy)
+
+    return HandledArgs(name, policy, workunit, view, initial_value)
+
+
+def get_annotations(parallel_type: str, handled_args: HandledArgs, *args, passed_kwargs) -> UpdatedTypes:
+    '''
+    parallel_type: A string identifying the type of parallel dispatch ("parallel_for", "parallel_reduce" ...)
+    handled_args: Processed arguments passed to the dispatch
+    args: raw arguments passed to the dispatch
+    passed_kwargs: raw keyword arguments passed to the dispatch
+
+    returns: UpdateTypes object or None if there are no annotations to be inferred
+
+    This function will infer first, the datatypes of policy arguments, and then any additional arguments
+    For readability: params/parameters will refer to the workunit signature
+                    args/argument will refer to the actual raw values passed during execution
+    '''
+
+    param_list = list(inspect.signature(handled_args.workunit).parameters.values())
+    args_list = list(*args)
+    updated_types = UpdatedTypes(workunit=handled_args.workunit, inferred_types={}, is_arg=set(), layout_change={})
+    policy_params: int = len(handled_args.policy.begin) if isinstance(handled_args.policy, MDRangePolicy) else 1
+
+    # accumulator 
+    if parallel_type == "parallel_reduce":
+        policy_params += 1
+    # accumulator + lass_pass
+    if parallel_type == "parallel_scan":
+        policy_params += 2
+
+    # Handling policy parameters
+    for i in range(policy_params):
+        param = param_list[i]
+
+        if param.annotation is inspect._empty:
+            # Check policy and apply annotation(s)
+            if isinstance(handled_args.policy, RangePolicy) or isinstance(handled_args.policy, TeamThreadRange):
+                # only expects one param
+                if i == 0:
+                    updated_types.inferred_types[param.name] = "int"
+                    updated_types.is_arg.add(param.name)
+
+            elif isinstance(handled_args.policy, TeamPolicy):
+                if i == 0:
+                    updated_types.inferred_types[param.name] = 'pk.TeamMember'
+                    updated_types.is_arg.add(param.name)
+
+            elif isinstance(handled_args.policy, MDRangePolicy):
+                total_dims = len(handled_args.policy.begin) 
+                if i < total_dims:
+                    updated_types.inferred_types[param.name] = "int"
+                    updated_types.is_arg.add(param.name)
+            else:
+                raise ValueError("Automatic annotations not supported for this policy")
+
+            # last policy param for parallel reduce and second last for parallel_scan is always the accumulator; the default type is double
+            if i == policy_params - 1 and parallel_type == "parallel_reduce" or i == policy_params - 2 and parallel_type == "parallel_scan":
+                updated_types.inferred_types[param.name] = "Acc:double"
```

Comment:
Above you use `pk.TeamMember`, but here you just use `Acc`, you should be more consistent. I think you can just remove `pk.` from `TeamMember` above right?

Reply:
True, I have removed the pk prefix